### PR TITLE
Highlighting the dynamics around what must be Opened

### PR DESCRIPTION
**Pull request by quaid**
- Moved and renamed a question on must-conditions so it contrasted
  with the question on should/recommended conditions.
- Providing structure that supports being able to perform a gap
  analysis and draw conclusions about which components need not or
  cannot be Opened.

Signed-off-by: Karsten Wade <kwade@redhat.com>

### CODE REVIEW

**Comment by quaid** [comment]
Tagging @stefwalter

These adjustments came from me looking to see if anything needed to be added in the language to clarify the tension around defining MUST items and what is optional. I found the content was mainly present in the FAQ, but not connected together physically, and the language used was disconnected.

File: open-source-services.md [hunk 1]
```diff
@@ -28,19 +28,47 @@ Thus we must address common questions about the requirements:
 
 ### Q: What defines a service?
 
-A service is software operated for the user. Software the user is operating for themself is not considered a service. A service is defined by the context in which the software is operated and all of the assets required to operate the software for the user in that context.
+A service is software operated for the user.
+Software the user is operating for themself is not considered a service.
+A service is defined by the context in which the software is operated and all of the assets required to operate the software for the user in that context.
```
Comment (stefwalter) [inline open-source-services.md:33]: This seems like simply a formatting changed mixed in with substantive changes. Did I understand that right?

File: open-source-services.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ This represents what we expect from any _service_ claiming to be Open Source.
 These requirements do not represent best practices, but are the bare minimum for Open Source development of services.
 Most Open Source models (beyond “throw it over the wall” behavior) are covered by these requirements in one way or another:
 
- 1. **All the service's code and assets necessary to operate the service are shared under an open source license, and publicly accessible.**
+ 1. **All the service's code and assets necessary to operate the service are shared under an Open Source license, and publicly accessible.**
```
Comment (stefwalter) [inline open-source-services.md:17]: Makes sense.

File: open-source-services.md [hunk 3]
```diff
@@ -28,19 +28,47 @@ Thus we must address common questions about the requirements:
 
 ### Q: What defines a service?
 
-A service is software operated for the user. Software the user is operating for themself is not considered a service. A service is defined by the context in which the software is operated and all of the assets required to operate the software for the user in that context.
+A service is software operated for the user.
+Software the user is operating for themself is not considered a service.
+A service is defined by the context in which the software is operated and all of the assets required to operate the software for the user in that context.
 
-### Q: What “assets” should be Open Source?
+This is a lightweight definition of service, read the Red Hat ["What are cloud services?"](https://www.redhat.com/en/topics/cloud-computing/what-are-cloud-services) article for a more broad discussion of cloud services.
```
Comment (stefwalter) [inline open-source-services.md:35]: Should we wrap the word service in quotes? Otherwise this reads oddly.

File: open-source-services.md [hunk 4]
```diff
@@ -28,19 +28,47 @@ Thus we must address common questions about the requirements:
...
+
+Where the [Open Source definition](https://opensource.org/osd) tells us how to identify a software code base as being Open Source, that definition does not cover what is needed to Open a service.
+
+For that we use the [Open Definition 2.1](https://opendefinition.org/od/2.1/en/), which starts with this summary of the definition:
+
+> _Knowledge is [O]pen if anyone is free to access, use, modify, and share it — subject, at most, to measures that preserve provenance and openness._
+
+For an Open Source code base to run as an Open Source service, there are a number of [Open Works](https://fossrit.github.io/open-work-definition/) around it needed to make it run in a cloud or service environment: scripts, documentation, policies, configurations, and so forth.
+
+### Q: What "assets and dependencies" should be Open Source?
```
Comment (stefwalter) [inline open-source-services.md:47]: Why was the word "dependencies" added here? This question is not about dependencies, rather the one below is.

File: open-source-services.md [hunk 5]
```diff
@@ -28,19 +28,47 @@ Thus we must address common questions about the requirements:
...
 A general rule of thumb is if you commit it to a repository, it should be open.
 Assets that should not be shared are ones where you don’t have redistribution rights (such as a RHEL VM image, or private information, and personal identifiable information), or things that are actually secret: Don’t share keys, passwords, or tokens.
 
 If changing configuration is an activity that a typical team member does in order to change the service, then you should share that configuration publicly, in order to meet **requirement #2**.
 
-It is _not requried_ that all tooling dependencies and service dependencies used during the building, testing, deployment and operating of the service are themselves Open Source.
+It is _not required_ that all tooling dependencies and service dependencies used during the building, testing, deployment, and operating of the service are themselves Open Source.
+
+It is _recommended_ that you include sufficient configuration to operate the service, or some base default version of configuration.
+Putting secrets in your configuration is a bad practice that should be avoided, regardless of Open Source.
```
Comment (stefwalter) [inline open-source-services.md:58]: This seems like simply a formatting changed mixed in with substantive changes. Did I understand that right?

File: open-source-services.md [hunk 6]
```diff
@@ -28,19 +28,47 @@ Thus we must address common questions about the requirements:
...
+
+### Q: What "assets and dependencies" MUST be Open too?
```
Comment (stefwalter) [inline open-source-services.md:60]: This question is about dependencies, why is the word assets also included here?

File: open-source-services.md [hunk 7]
```diff
@@ -28,19 +28,47 @@ Thus we must address common questions about the requirements:
...
+**A:** It is _not required_ that all dependencies of a service are Open Source, or have Open Source alternatives.
+For example, a service may interact with a proprietary authentication system, or run on a proprietary platform.
+
+The requirements together serve to enable others to change the service code so that its proprietary dependencies can be replaced with Open Source ones, should that be desired.
+
+The best approach to take depends on if you are working with a new service or an existing service:
+1. For a new service, the "Open by default" practice means you limit your use of private code or non-Open services, platforms, etc.
+If you do have to choose a private/non-Open solution, make it a deliberate choice, such as when it is required in the environment where you build or run the service.
+2. For an existing service, you need to do a gap analysis to discover what assets and dependencies are private/non-Open.
+Then determine for each asset or dependency if it is necessary to the service being able to run in another environment
```
Comment (stefwalter) [inline open-source-services.md:71]: Why are these latter steps necessary? Running in another environment, while desirable attribute of many services, is not one of these requirements.

File: open-source-services.md [hunk 8]
```diff
@@ -73,19 +101,12 @@ However, if a typical team member operates their own instance of the service in
 It is a powerful, laudable goal for others to be able to operate an Open Source service, either by operating an instance on their own, or participating in operations of a common instance.
 [Operate First](https://www.operate-first.cloud/) is a project that enables this, and even provides infrastructure.
 
-### Q: Must all dependencies for a service be Open Source too?
-
-**A:** It is _not required_ that all dependencies of a service are Open Source, or have Open Source alternatives.
-For example, a service may interact with a proprietary authentication system, or run on a proprietary platform.
-
-The requirements together serve to enable others to change the service code so that its proprietary dependencies can be replaced with Open Source ones, should that be desired.
-
```
Comment (stefwalter) [inline open-source-services.md:82]: This was moved above, and the FAQ question was changed to include assets. While I'm fine with moving the two questions one after another, I believe the topic of assets and dependencies should have separate questions even if the answers refer to one another in subtle ways.

File: open-source-services.md [hunk 9]
```diff
@@ -73,19 +101,12 @@ However, if a typical team member operates their own instance of the service in
...
 ### Q: Must there be a community for the service?
 
-**A:** The minimal requirements above _do not require_ that the Open Source service lead a community built around it.
+**A:** The minimal requirements above _do not require_ that the Open Source service create or lead a community built around it.
```
Comment (stefwalter) [inline open-source-services.md:106]: Works for me.

File: open-source-services.md [hunk 10]
```diff
@@ -73,19 +101,12 @@ However, if a typical team member operates their own instance of the service in
...
 
 If your project grows bigger, and has many participants, then forming a community with the typical components (communication venues, code of conduct, governance, etc.) is _recommended_, and brings significant further Open Source benefits.
 
 Among other things, having a community means sharing your documentation, issues, and a roadmap. It also means conducting most planning conversations about the service in the open, and not limiting participation to your team.
 This is how you gain the advantages of the Open innovation process, where minds beyond your team can help work through problems and create solutions
-Without these things, your community interaction will be more painful as crossing the boundary between community and Red Hat private has a high context switching cost.
+Without these things, your community interaction will be more painful as crossing the boundary between community and Red Hat private has a high context-switching cost.
```
Comment (stefwalter) [inline open-source-services.md:112]: Nice.

**Review by stefwalter — CHANGES_REQUESTED** [review]
Lots of great additions!

However I believe the FAQuestions around dependencies and assets should remain distinct ... even if their answers refer to one another subtly for clarity. 


Sorry for the delay in review. It took a while longer to review, because this mixed in syntax formatting changes with substantive changes.

File: open-source-services.md (hunk 1 above)
Comment (quaid) [reply open-source-services.md:33]: It is only a formatting change, I don't think there is a change in the content itself.

In fact, the reason for the formatting change is to make it easier to read substantive changes.

File: open-source-services.md (hunk 2 above)
Comment (quaid) [reply open-source-services.md:17]: Yeah, consistency, and I've moved to using Open whenever it's the case.

File: open-source-services.md (hunk 3 above)
Comment (quaid) [reply open-source-services.md:35]: The challenge is, in English the use of "quote marks" can mean a lack of confidence about the words. But you are correct that we do want to emphasize it, which is typically done with italics. I'll make that suggestion into this one.

**Comment by stefwalter** [comment]
@quaid When you're back next week. Many of these changes can be brought in, but some need reworking. Should we split them apart so that some changes can be merged?

**Comment by stefwalter** [comment]
@quaid Should I pick some changes from the big ball to help get this untangled? Like formatting changes for example?

**Pull request closed without merging**